h5glance/utils.py

```diff
@@ -0,0 +1,48 @@
+"""Helper functions
+"""
+import h5py
+
+_mapping = {}
+_mapping[h5py.File] = "file"
+_mapping[h5py.Group] = "group"
+_mapping[h5py.Dataset] = "dataset"
+
+
+def register_h5pyd():
+    import h5pyd
+    _mapping[h5pyd.File] = "file"
+    _mapping[h5pyd.Group] = "group"
+    _mapping[h5pyd.Dataset] = "dataset"
+
+
+def check_class(obj_class):
+    """Check registered class"""
+    lib = obj_class.__module__.split(".")[0]
+    if lib == "h5pyd":
+        register_h5pyd()
+    else:
+        _mapping[obj_class] = None
+
+
+def is_file(obj):
+    """Returns true if the object is a h5py-like file."""
+    obj_class = obj.__class__
```

Comment:
I think the idiomatic way to get the class of an object is now `type(obj)`.

Reply:
It's up to you.

I update the code and create a function `get_h5py_kind` to make it simple.